### PR TITLE
John/create listing form improvements

### PR DESCRIPTION
**Pull request by jordajm**
### Checklist:

- [x] Test your work and double-check to confirm that you didn't break anything

### Description:
Ticket: https://github.com/OriginProtocol/origin-dapp/issues/432
- Creates price-field widget that works with react-jsonschema-form
- Adds photo-picker widget with image previews
- Replaces several price conversion functions with one `getFiatPrice` util that can be reused across the dapp
- Adds ETH icon in price field
- Removes old unused util `src/utils/numberFormat.js`
- Makes description field into text area

### Price field widget:
screenshot 2018-08-28 09 20 24

### Photo picker widget:
screenshot 2018-08-28 01 13 14
screenshot 2018-08-28 01 13 26

### CODE REVIEW

File: src/components/form-widgets/price-field.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        price: value,
+        priceUsd
+      }, () => this.props.onChange(value))
+    }
+  }
+
+  render() {
+    return(
+      this.priceHidden ?
+        <div></div> :
```
Comment (micahalcorn) [inline src/components/form-widgets/price-field.js]: Any reason to return this empty div rather than `null`?

Comment (jordajm) [reply src/components/form-widgets/price-field.js]: Good point! Pushing a change to `token-ui` branch

File: src/components/listing-card-prices.js [hunk 2]
```diff
@@ -27,76 +31,20 @@ class ListingCardPrices extends Component {
...
-  }
-
   render() {
     return (
       <div>
         <div className="d-flex align-items-center price-container">
           <div>
             <div className="d-inline-block price placehold">
-              {this.state.exchangeRate == null && (
+              {this.state.fiatPrice == null && (
```
Comment (micahalcorn) [inline src/components/listing-card-prices.js]: Any reason to use `==` vs `===` here?

Comment (jordajm) [reply src/components/listing-card-prices.js]: ahh yeah - this was already there and I didn't notice it... pushed a change to `token-ui` to fix it

**Pull request merged**